homeassistant/components/songpal/media_player.py

```diff
@@ -65,33 +63,39 @@
 )
 
 
-async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
-    """Set up the Songpal platform."""
-    if PLATFORM not in hass.data:
-        hass.data[PLATFORM] = {}
+async def async_setup_platform(
```

Comment:
To add config flow, the integration can't use platform set up and should be set up as a top level config entry. You can see how I implemented this for `vizio` here: https://github.com/home-assistant/core/blob/2e6edb0447f6e6c460eb5ab1a1d9c75602eebddd/homeassistant/components/vizio/__init__.py
https://github.com/home-assistant/core/blob/2e6edb0447f6e6c460eb5ab1a1d9c75602eebddd/homeassistant/components/vizio/media_player.py

Reply:
This function can be removed.